### ISSUE TITLE
Problem with flexible array members in OnlTools unions

### ISSUE THREAD

**Issue by plexoos** [star-bnl/star-sw#368]
Flexible array members and zero size arrays are not allowed to be members of unions according to C++. Gcc 11 reveals this problem in the code and here is an example error in OnlTools/Jevp/StJevpBuilders/bbcBuilder.h

```
g++ -fPIC -pipe -Wall -Woverloaded-virtual -std=c++0x -Wno-long-long -pthread -Wno-deprecated-declarations -m64 -O2 -g -falign-loops -falign-jumps -falign-functions -Dsl79_gcc11 -D__ROOT__ -DQT_THREAD_SUPPORT -DQT_SHARED -DQT_NO_DEBUG -DNEW_DAQ_READER -I. -IOnlTools -IStRoot/RTS/src -IStRoot/RTS/include -IStRoot/RTS/trg/include -I. -IStRoot -I.sl79_gcc11/include -IOnlTools/Jevp/StJevpPlot -IOnlTools/Jevp/StJevpPresenter -IOnlTools/Jevp/StJevpServer -IOnlTools/Jevp/StJevpData -IOnlTools/Jevp/StJevpBuilders -IOnlTools/Jevp/StJevpViewer -I. -IOnlTools -IStRoot/RTS/src -IStRoot/RTS/include -IStRoot/RTS/trg/include -I. -IStRoot -I.sl79_gcc11/include -I/opt/software/linux-scientific7-x86_64/gcc-11.2.1/root-5.34.38-p7sjrh5whm6m2ao3zxkaoaqomiscvw52/include -I/opt/software/linux-scientific7-x86_64/gcc-11.2.1/root-5.34.38-p7sjrh5whm6m2ao3zxkaoaqomiscvw52/include -I/usr/include/mysql -I/opt/software/linux-scientific7-x86_64/gcc-11.2.1/libxml2-2.9.12-k3pblgujcxjatlcsu2xmhe7qcwtqpxn4/include/libxml2 -I/opt/software/linux-scientific7-x86_64/gcc-11.2.1/libiconv-1.16-ghg7qgnp355e4h2zsuua7ar3ngindtwn/include -c .sl79_gcc11/OBJ/OnlTools/Jevp/StJevpBuilders/bbcBuilder.cxx -o .sl79_gcc11/OBJ/OnlTools/Jevp/StJevpBuilders/bbcBuilder.o
In file included from .sl79_gcc11/OBJ/OnlTools/Jevp/StJevpBuilders/bbcBuilder.cxx:15:
.sl79_gcc11/OBJ/OnlTools/Jevp/StJevpBuilders/bbcBuilder.h:30:10: error: flexible array member in union
   30 |     TH1 *array[];
      |          ^~~~~
cons: *** [.sl79_gcc11/OBJ/OnlTools/Jevp/StJevpBuilders/bbcBuilder.o] Error 1
cons: errors constructing .sl79_gcc11/OBJ/OnlTools/Jevp/StJevpBuilders/bbcBuilder.o
```

**Comment by plexoos** [comment]
A fix like this seems to silence the compiler:

```diff
-TH1 *array[];
+TH1 *array[1];
```

https://github.com/star-bnl/star-sw/pull/366/commits/56aa19aa05a7fe8a27e7c7dac0562523ef12dad2

@jml985 Do you have a better solution?

**Comment by jml985** [comment]
I'll check later this afternoon.

**Comment by jml985** [comment]
@plexoos I put in a pull request #369 updating all of the builders according to your suggestion.    This works fine with the use.

**Comment by plexoos** [comment]
Thank you Jeff for looking into this. I think in your last commit on [jml985:JevpProduction](https://github.com/jml985/star-sw/tree/JevpProduction) you missed a few builders. If you look at my commit https://github.com/star-bnl/star-sw/pull/366/commits/56aa19aa05a7fe8a27e7c7dac0562523ef12dad2 there are a few builders which use a different name for the array of histogram pointers, e.g. `TH2 *tbVsAdcArray[];` instead of `TH1 *array[];` as in my example above. I don't think I missed anything because the code actually compiles with gcc 11 and the errors are not reported anymore. In the log message I documented the command I came up with to do the substitution, it can be used to recreate the patch.
